taichi/codegen/spirv/spirv_codegen.cpp

```diff
@@ -516,18 +516,24 @@ class TaskCodegen : public IRVisitor {
   }
 
   void visit(ReturnStmt *stmt) override {
+    // Now we only support one ret
+    const auto &ret_attribs = ctx_attribs_->rets()[0];
+    auto dt = ret_attribs.dt;
+    if (auto tensor_type = dt->cast<TensorType>()) {
+      dt = tensor_type->get_element_type();
+    }
     for (int i = 0; i < stmt->values.size(); i++) {
-      const auto &ret_attribs = ctx_attribs_->rets()[i];
-      const auto dt = ret_attribs.dt;
       spirv::Value buffer_val = ir_->make_value(
           spv::OpAccessChain,
           ir_->get_storage_pointer_type(ir_->get_primitive_type(dt)),
           get_buffer_value(BufferType::Rets, PrimitiveType::i32),
-          ir_->int_immediate_number(ir_->i32_type(), i));
+          ir_->int_immediate_number(ir_->i32_type(), (i << 1)));
```

Comment:
This statement is getting a pointer to the struct member variable. Should not shift here, but rather use the correct pointer types

Reply:
fix